### PR TITLE
HS-782 Add delete minion feature

### PR DESCRIPTION
**Pull request by chinh-cle**
## Jira link(s)
- https://issues.opennms.org/browse/HS-782-minion-delete

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: ui/src/components/Appliances/AppliancesMinionsTable.vue [hunk 1]
```diff
@@ -31,17 +32,39 @@
...
           </tr>
         </TransitionGroup>
       </table>
     </div>
   </TableCard>
-  <PrimaryModal :visible="modal.isVisible" :title="modal.title" :hide-title="modal.hideTitle">
+  <PrimaryModal :visible="isVisible" :title="modal.title" :class="modal.cssClass">
     <template #content>
-      <LineGraph :graph="graphProps" />
+      <div v-html="modal.content"></div>
```
Comment (mikewrosey) [inline ui/src/components/Appliances/AppliancesMinionsTable.vue:52]: this v-html is causing the sonar security hotspot.. could we change it to `<p>{{ modal.content }}</p>` ?

File: ui/src/components/Appliances/AppliancesMinionsTable.vue [hunk 2]
```diff
@@ -50,30 +73,72 @@
...
+  }
+}
+const onDelete = (id: string, label: string) => {
   modal.value = {
     ...modal.value,
-    isVisible: true
+    title: label || '',
+    cssClass: 'modal-delete-minion',
+    content: `
+      <p>Do you want to delete</p>
```
Comment (mikewrosey) [inline ui/src/components/Appliances/AppliancesMinionsTable.vue:133]: if not using the v-html, we can remove the paragraph tags. Also probably want a question mark at the end.

File: ui/src/components/Common/PrimaryModal.vue [hunk 3]
```diff
@@ -47,3 +47,21 @@ watchEffect(() => {
   position: relative;
 }
 </style>
+
+<style lang="scss">
+// TODO: need to find a way to scope the style below. ':deep' selector does not seem to work
+@use "@featherds/styles/themes/variables";
+.modal-delete-minion {
```
Comment (mikewrosey) [inline ui/src/components/Common/PrimaryModal.vue:54]: Presumably we want to use this class for more than just the minion, and this is a common component so
maybe we can rename it to just `modal-delete`.

File: ui/src/store/Mutations/minionMutations.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+import { DeleteMinionDocument } from '@/types/graphql'
+
+const { startSpinner, stopSpinner } = useSpinner()
+
+export const useMinionMutations = defineStore('minionMutations', () => {
+  const {
+    execute: deleteMinion,
+    isFetching: isDeletingMinion,
+    error: isDeleteMinionError
```
Comment (mikewrosey) [inline ui/src/store/Mutations/minionMutations.ts:13]: I think this line can be removed as I don't see the error used. Our GQL interceptor should pick up any errors and display automatically.

**Review by anyarybalova — APPROVED** [review]

**Pull request merged**